pages/learn/json-schema-examples.md

```diff
@@ -92,129 +177,403 @@ title: JSON Schema examples
 }
 ```
 
-## Card
+**Data**
+
 ```json
 {
-  "$id": "https://example.com/card.schema.json",
+  "startDate": "2023-08-25T10:00:00Z",
+  "endDate": "2023-08-25T12:00:00Z",
+  "summary": "Conference Presentation",
+  "location": "Conference Center",
+  "recurrenceDule": "FREQ=DAILY;COUNT=5"
+}
+```
+
+
+## Device type
+
+This schema represents electronic devices with a `deviceType` property that determines the device's category, such as `Smartphone` or `Laptop`.It employs the `$dynamicRef` keyword to dynamically reference schemas based on the `deviceType` property. This flexible schema structure allows data to conform to the appropriate device schema based on the `deviceType` specified, making it easy to describe different electronic devices with their unique characteristics
+
+```json
+{
+  "$id": "https://example.com/device.schema.json",
   "$schema": "https://json-schema.org/draft/2020-12/schema",
-  "description": "A representation of a person, company, organization, or place",
   "type": "object",
-  "required": [ "familyName", "givenName" ],
   "properties": {
-    "fn": {
-      "description": "Formatted Name",
+    "deviceType": {
+      "type": "string"
+    }
+  },
+  "required": ["deviceType"],
+  "$dynamicRef": "#/$defs/DeviceTypes"
+}
+
+{
+  "$id": "https://example.com/smartphone.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "type": "object",
+  "properties": {
+    "brand": {
+      "type": "string"
+    },
+    "model": {
       "type": "string"
     },
-    "familyName": {
+    "screenSize": {
+      "type": "number"
+    }
+  },
+  "required": ["brand", "model", "screenSize"]
+}
+
+{
+  "$id": "https://example.com/laptop.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "type": "object",
+  "properties": {
+    "brand": {
+      "type": "string"
+    },
+    "model": {
+      "type": "string"
+    },
+    "processor": {
+      "type": "string"
+    },
+    "ramSize": {
+      "type": "number"
+    }
+  },
+  "required": ["brand", "model", "processor", "ramSize"]
+}
+```
+
+**Data**
+
+```json
+{
+  "deviceType": "Smartphone",
+  "brand": "Samsung",
+  "model": "Galaxy S21",
+  "screenSize": 6.2
+}
+```
+
+
+## Ecommerce system
+
+A schema representing an ecommerce system, where `$anchor` is used within the definitions of `product` and `order` schemas to define anchor points: `ProductSchema` and `OrderSchema`, respectively.
+
+```json
+{
+  "$id": "https://example.com/ecommerce.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$defs": {
+    "product": {
+      "$anchor": "ProductSchema",
+      "type": "object",
+      "properties": {
+        "name": { "type": "string" },
+        "price": { "type": "number", "minimum": 0 }
+      }
+    },
+    "order": {
+      "$anchor": "OrderSchema",
```

Comment:
This anchor isn't strictly necessary since it's not used anywhere, but leaving it isn't hurting anything.  (It might be used externally.)